test/framework/src/main/java/org/elasticsearch/transport/nio/channel/ChannelFactory.java

```diff
@@ -39,23 +38,27 @@
 
 public class ChannelFactory {
 
-    private final TcpReadHandler handler;
+    private final Consumer<NioSocketChannel> contextSetter;
     private final RawChannelFactory rawChannelFactory;
 
-    public ChannelFactory(TcpTransport.ProfileSettings profileSettings, TcpReadHandler handler) {
-        this(new RawChannelFactory(profileSettings), handler);
+    public ChannelFactory(TcpTransport.ProfileSettings profileSettings, Consumer<NioSocketChannel> contextSetter) {
```

Comment:
Would you add a Javadoc for this constructor explaining the purpose of the `contextSetter` parameter?